<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Genomics context of missense variants &mdash; NAIC Genomics training sessions 0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
      <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
      <link rel="stylesheet" type="text/css" href="../../_static/sphinx_lesson.css?v=0c089442" />
      <link rel="stylesheet" type="text/css" href="../../_static/tabs.css?v=a5c4661c" />
      <link rel="stylesheet" type="text/css" href="../../_static/tabs.css?v=a5c4661c" />

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../_static/documentation_options.js?v=2709fde1"></script>
        <script src="../../_static/doctools.js?v=888ff710"></script>
        <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../../_static/copybutton.js?v=f281be69"></script>
        <script src="../../_static/minipres.js?v=a0d29692"></script>
        <script>let toggleHintShow = 'Click to show';</script>
        <script>let toggleHintHide = 'Click to hide';</script>
        <script>let toggleOpenOnPrint = 'true';</script>
        <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
        <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
        <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
        <script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
        <script>mermaid.initialize({startOnLoad:true});</script>
        <script src="../../_static/mermaid.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            NAIC Genomics training sessions
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../ngs-gpu.html">Accelerated NGS analysis in AI cloud (GPU cloud)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Alphamissense.html">Deep learning use case in genomics</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">NAIC Genomics training sessions</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Genomics context of missense variants</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/episodes/AlphaMissense/01_genomics_context.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="genomics-context-of-missense-variants">
<h1>Genomics context of missense variants<a class="headerlink" href="#genomics-context-of-missense-variants" title="Link to this heading"></a></h1>
<ul class="simple">
<li><p>Missense variants alter the amino acid sequence that could affect the
protein structure potentially leading to diseases</p></li>
<li><p>Missense variants leading to diseases are known as pathogenic
missense variants</p></li>
<li><p>Missense variants with limited effects not leading to diseases are
known as bening missense variants</p></li>
</ul>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p class="rubric">Challenges in the current clinical setting</p>
<ul class="simple">
<li><p>Understanding the pathogenicity (or bening status) of missense
variants is a major challenge in the current clinical setting</p></li>
<li><p>Over the years, scientists have identified a long list of disease
associated genes.</p>
<ul>
<li><p>Differentiating pathogenic and bening missense variants in
these set of genes is also a challenging task</p></li>
</ul>
</li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p class="rubric" id="current-methods-to-predict-the-pathogenicity-of-missense-variants-mega">Current methods to predict the pathogenicity of missense
variants</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Method</p></th>
<th class="head"><p>Tool</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Evolutionary conservation</p></td>
<td><p>ConSurf, FATHMM,
MutationAssessor,
PANTHER, PhD-SNP, SIFT,
SNPs&amp;GO</p></td>
</tr>
<tr class="row-odd"><td><p>Protein structure/function and
evolutionary conservation</p></td>
<td><p>Align GVGD, MAPP,
MutationTaster, MutPred,
PolyPhen-2</p></td>
</tr>
</tbody>
</table>
<p class="rubric" id="ensembl-variation-pathogenicity-predictions">Ensembl Variation pathogenicity predictions</p>
<p>Ref:
<a class="reference external" href="https://www.ensembl.org/info/genome/variation/prediction/protein_function.html">https://www.ensembl.org/info/genome/variation/prediction/protein_function.html</a></p>
<p><img alt="image1" src="https://md.sigma2.no/uploads/71e7a315-75f5-44d0-9669-be2af0a2da83.png" /></p>
<ul class="simple">
<li><p><strong>REVEL</strong>: Ensemble method that integrates scores from multiple
pathogenicity predicting tools to one score</p></li>
</ul>
<p class="rubric" id="limitations-in-current-methods">Limitations in current methods</p>
<ul class="simple">
<li><p>Current tools rely on applying hard-thresholds (cut-off scores)</p></li>
<li><p>Practically, these hard-thresholds are used only as guides</p>
<ul>
<li><p>For example, VEP user manual strongly recommends using an
appropriate cut-off depending on the experiment and dataset.</p></li>
</ul>
</li>
<li><p>These tools are used to predict the pathogenicity of other variant
types (not only missense)</p>
<ul>
<li><p>Tools have shown stronger pathogenicity scores for more harmful
variant types (stop-gain, stop-loss, splice-site) compared to
missense variants</p></li>
</ul>
</li>
</ul>
</div>
<div class="admonition danger">
<p class="admonition-title">Danger</p>
<p class="rubric" id="pathogenicity-of-missense-variants-fire">Pathogenicity of missense variants</p>
<ul class="simple">
<li><p>Predicting the pathogenicity of missense variants remains a
challenging task.</p></li>
<li><p>A large majority of currently identified missense variants fall to
the category - <em>variants with uncertain significance</em></p>
<ul>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7334197">doi: 10.1038/s41586-020-2308-7 (Nature. 2020; 581(7809):
434–443.)</a></p></li>
</ul>
</li>
</ul>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p class="rubric" id="deep-learning-based-solution-tada">Deep learning based solution</p>
<p><img alt="image2" src="https://md.sigma2.no/uploads/131d3219-a641-4fcb-9d2d-171006bb9614.png" /></p>
</div>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, NAIC.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>