<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Model training and validation &mdash; NAIC Genomics training sessions 0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
      <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
      <link rel="stylesheet" type="text/css" href="../../_static/sphinx_lesson.css?v=0c089442" />
      <link rel="stylesheet" type="text/css" href="../../_static/tabs.css?v=a5c4661c" />
      <link rel="stylesheet" type="text/css" href="../../_static/tabs.css?v=a5c4661c" />

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../_static/documentation_options.js?v=2709fde1"></script>
        <script src="../../_static/doctools.js?v=888ff710"></script>
        <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
        <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
        <script src="../../_static/copybutton.js?v=f281be69"></script>
        <script src="../../_static/minipres.js?v=a0d29692"></script>
        <script>let toggleHintShow = 'Click to show';</script>
        <script>let toggleHintHide = 'Click to hide';</script>
        <script>let toggleOpenOnPrint = 'true';</script>
        <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
        <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
        <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
        <script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
        <script>mermaid.initialize({startOnLoad:true});</script>
        <script src="../../_static/mermaid.js"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            NAIC Genomics training sessions
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../ngs-gpu.html">Accelerated NGS analysis in AI cloud (GPU cloud)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Alphamissense.html">Deep learning use case in genomics</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">NAIC Genomics training sessions</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Model training and validation</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../_sources/episodes/AlphaMissense/06_Model_training.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="model-training-and-validation">
<h1>Model training and validation<a class="headerlink" href="#model-training-and-validation" title="Link to this heading">ÔÉÅ</a></h1>
<p>This step use AlphaFold model, which is already trained to predict
protein structures <em>(pre-trained)</em>. Then this pre-trained model is
trained on a dataset prepared in the previous step.</p>
<div class="mermaid">
            graph TD;
i1(Benign &amp; Pathogenic - training data) --&gt; 1
i3(Benign &amp; Pathogenic - training data) --&gt; 6
i4(Balanced ClinVar &amp; benign variants) --&gt; 7
1(Updated AlphaFold pre-trained model)
1 --&gt; 2(Makes predictions for each variant)
2 --&gt; 3(Compare prediction to actual variants and calculate the error)
3 --&gt; 4(Update model parameters based on error)
4 --&gt; 5(Improved model)
5 --&gt; 6(Make predictions &amp; evaluate the performance)
6 --&gt; 4
6 --&gt; 7(Model validation)
7 --&gt; 4
7 --&gt; 8(Prediction performance reaches a stable level)
   classDef highlight fill:#99ccff;
   classDef white fill:#ffffff;
   class 1,2,3,4,5,6,7,8 highlight;
        </div><div class="admonition note">
<p class="admonition-title">Note</p>
<p class="rubric">Bening labelled variants:</p>
<ul class="simple">
<li><p>Human variants: gnomAD</p></li>
<li><p>Primate variants: Great Ape project and a few other sources</p></li>
<li><p>Remove training variants at protein positions if they appear on validation or test sets</p></li>
</ul>
<p class="rubric">Low-frequency variants are likely to be pathogenic than the frequent ones</p>
<ul class="simple">
<li><p>Used machine learning technique that adjusts model parameters
according to the variant frequency so that the <em>frequent variants
contribute more strongly to the training signal</em></p></li>
</ul>
<p class="rubric">Pathogenic labelled variants</p>
<ul class="simple">
<li><p>Unobserved in human and primate populations (unobserved data)</p></li>
</ul>
<p class="rubric">Training data</p>
<ul class="simple">
<li><p>Balanced dataset reflecting the true distribution of pathogenic and benign missense variants in the population</p></li>
<li><p>The pathogenic set contains the same number of variants for each trinucleotide context as the benign set</p></li>
<li><p>Probability of sampling a pathogenic variant depends on abundance of its protein in the benign set</p></li>
<li><p>Variants in proteins that play a role in critical cellular function are likely to be pathogenic. Minimize the bias of functional context of protein in training set</p></li>
</ul>
<p class="rubric">Effects of using balanced training and test dataset</p>
<ul class="simple">
<li><p>Unobserved data (pathogenic variants in training data) can have likely bening variants. Following technique can be used to filter-out likely bening variants in unobserved data</p></li>
</ul>
<ol class="arabic simple">
<li><p>Use <code class="docutils literal notranslate"><span class="pre">AlphaMissense</span> <span class="pre">initial</span> <span class="pre">model</span></code> - Already learned to
differentiate pathogenic and benign variants</p></li>
<li><p>Predict pathogenicity of unobserved variants using initial model</p></li>
<li><p>Use new scores to remove likely bening variants in unobserved data</p></li>
</ol>
<p>Minimize the probability of including likely bening unobserved variants in downstream AlphaMissense model training rounds</p>
</div>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, NAIC.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>